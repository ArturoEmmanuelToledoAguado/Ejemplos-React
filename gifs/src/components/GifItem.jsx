import PropTypes from 'prop-types';

export const GifItem = ({title,url}) => {
    const copyPath = ()=>{
        navigator.clipboard.writeText(url)
    }

  return (
    <div className="">
        <img 
            src={url}
            alt={title}
        />
        <p>{title}</p>
        <button className=""onClick={copyPath}>Copy Path</button>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}