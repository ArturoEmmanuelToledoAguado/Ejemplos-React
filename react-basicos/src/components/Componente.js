import React from "react";
//import React,{Component} from "react";

const Componente=props=> <h2>{props.msg}</h2>

/* Componente en funcion declarada
function Componente(props) {
    return <h2>{props.msg}</h2>;
}*/

/* Componente con clase
class Componente extends Component{
    render() {
        return <h2>{this.props.msg}</h2>;
    }
}*/

export default Componente;