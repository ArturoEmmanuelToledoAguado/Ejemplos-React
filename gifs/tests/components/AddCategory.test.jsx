import { render, screen, fireEvent } from "@testing-library/react"
import { AddCategory } from "../../src/components/AddCategory";

describe('prueba sobre AddCategory', () => { 
    test('Prueba del snapshot', () => { 
        const {container} = render(<AddCategory onNewCategory={()=>{}}/>)
        expect(container).toMatchSnapshot();
     })

    test('Prueba para escribir en el Input', () => { 
        render(<AddCategory onNewCategory={()=>{}}/>)
        const input = screen.getByRole('textbox')
        fireEvent.input(input, {target:{value:'hola'}})
        expect(input.value).toEqual('hola')
     })

    test('Validar el funcionamiento del formulario', () => { 
        const onNewCategory = jest.fn()
        render(<AddCategory onNewCategory={onNewCategory}/>)
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        fireEvent.input(input, {target:{value:'algo'}})
        fireEvent.submit(form)
        expect(input.value).toEqual('')
        expect(onNewCategory).toHaveBeenCalledWith('algo')
     })
})