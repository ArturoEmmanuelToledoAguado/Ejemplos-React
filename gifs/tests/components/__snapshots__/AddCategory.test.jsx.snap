// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba sobre AddCategory Prueba del snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Write Something"
      type="text"
      value=""
    />
  </form>
</div>
`;
